Sources/Apodini/Request/Request.swift

```diff
@@ -6,29 +6,151 @@
 //
 import Foundation
 import NIO
-import protocol Fluent.Database
+import protocol FluentKit.Database
+import struct FluentKit.DatabaseID
+
+protocol ApodiniRequest: CustomStringConvertible, CustomDebugStringConvertible {
+    /// Returns a description of the Apodini Request.
+    /// If the `ExporterRequest` also conforms to `CustomStringConvertible`, its `description`
+    /// will be appended.
+    var description: String { get }
+    /// Returns a debug description of the Apodini Request.
+    /// If the `ExporterRequest` also conforms to `CustomDebugStringConvertible`, its `debugDescription`
+    /// will be appended.
+    var debugDescription: String { get }
+
+    var endpoint: AnyEndpoint { get }
 
-/// A type representing a server response.
-public protocol Request: CustomStringConvertible {
-    /// parameters
-    func parameter<T: Codable>(for parameter: UUID) throws -> T?
-    /// The EventLoop associated with this request
     var eventLoop: EventLoop { get }
-    /// database
-    var database: Fluent.Database? { get }
+
+    var defaultDatabase: Database? { get }
+    func database(_ id: DatabaseID?) -> Database?
+
+    func retrieveParameter<Element: Codable>(_ parameter: Parameter<Element>, of element: Element.Type) throws -> Element
+
+    func enterRequestContext<E, R>(with element: E, executing method: (E) -> EventLoopFuture<R>) -> EventLoopFuture<R>
+    func enterRequestContext<E, R>(with element: E, executing method: (E) -> R) -> R
+}
+
+struct Request<I: InterfaceExporter, C: Component>: ApodiniRequest {
+    var description: String {
+        var description = "Apodini Request:\n"
+        if let convertible = exporterRequest as? CustomStringConvertible {
+            description += convertible.description
+        }
+        return description
+    }
+    var debugDescription: String {
+        var debugDescription = "Apodini Request:\n"
+        if let convertible = exporterRequest as? CustomDebugStringConvertible {
+            debugDescription += convertible.debugDescription
+        }
+        return debugDescription
+    }
+
+    var exporter: I
+    var exporterRequest: I.ExporterRequest
+
+    var storedEndpoint: Endpoint<C>
+    var endpoint: AnyEndpoint {
+        storedEndpoint
+    }
+
+    var eventLoop: EventLoop
+
+    var defaultDatabase: Database? {
+        database(nil)
+    }
+    /// Use to inject a database in our test cases
+    var databaseRetrieval: ((DatabaseID?) -> Database)?
```

Comment:
wouldn't it be better to create a dedicated mock request struct for our tests?

Reply:
Well yes, as the whole database stuff is being reworked anyways I thought this kind of "fast and dirty" way will do for now.

Reply:
Obsolet as of 45f826d